pandas/tests/extension/test_arrow.py

```diff
@@ -3246,6 +3246,14 @@ def test_arrow_floordiv_large_values():
     tm.assert_series_equal(result, expected)
 
 
+def test_arrow_floordiv_large_integral_result():
```

Comment:
Not sure if a whatsnew is needed, it seems very similar to one added by https://github.com/pandas-dev/pandas/pull/56647

Reply:
Added one since issue is slightly different.

Reply:
Nvm, removed it because the issue only exists on main, this PR is really a followup to https://github.com/pandas-dev/pandas/pull/56647 which hasn't been released yet.